src/lib/storage/frame_of_reference_column.hpp

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+#include <boost/hana/contains.hpp>
+#include <boost/hana/tuple.hpp>
+#include <boost/hana/type.hpp>
+
+#include <type_traits>
+
+#include <array>
+#include <memory>
+
+#include "base_encoded_column.hpp"
+#include "types.hpp"
+
+namespace opossum {
+
+class BaseCompressedVector;
+
+/**
+ * @brief Column implementing frame-of-reference encoding
+ */
+template <typename T, typename = std::enable_if_t<encoding_supports(enum_c<EncodingType, EncodingType::FrameOfReference>, hana::type_c<T>)>>
```

Comment:
Please don’t mind this monstrous `enable_if` here. I tried to simplify it but GCC tried to eat me alive. 

Ideally this should be something like (clang at least was fine with it):
```c++
std::enable_if_t<encoding_supports<EncodingType::FrameOfReference, T>>
```